lib/addons/karpenter/index.ts

```diff
@@ -686,40 +701,57 @@ export class KarpenterAddOn extends HelmAddOn {
      * Outputs to CloudFormation and map the role to the aws-auth ConfigMap
      * @param cluster EKS Cluster
      * @param stackName Name of the stack
+     * @param nodeRoleAdditionalPoliciesDir Directory containing additional policies to be attached to the Node Role
      * @param region Region of the stack
      * @returns [karpenterNodeRole, karpenterInstanceProfile]
      */
-    private setUpNodeRole(cluster: Cluster, stackName: string, region: string): [iam.Role, iam.CfnInstanceProfile] {
+    private setUpNodeRole(clusterInfo: ClusterInfo, stackName: string, nodeRoleAdditionalPoliciesDir: string | undefined, region: string): [iam.Role, iam.CfnInstanceProfile] {
+        
+        const cluster = clusterInfo.cluster as Cluster;
+        
         // Set up Node Role
-        const karpenterNodeRole = new iam.Role(cluster, 'karpenter-node-role', {
-            assumedBy: new iam.ServicePrincipal(`ec2.${cluster.stack.urlSuffix}`),
-            managedPolicies: [
-                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSWorkerNodePolicy"),
-                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKS_CNI_Policy"),
-                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEC2ContainerRegistryReadOnly"),
-                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonSSMManagedInstanceCore"),
-            ],
+        const karpenterNodeRole = new iam.Role(clusterInfo.cluster, 'karpenter-node-role', {
+            assumedBy: new iam.ServicePrincipal(`ec2.${clusterInfo.cluster.stack.urlSuffix}`),
+            managedPolicies: DEFAULT_KARPENTER_NODE_ROLE_POLICIES,
```

Comment:
Can we allow adding to the list of managed policies here, via the config of the addon?
eg.  When using the CloudwatchInsights addon, you need to add:
```
    iam.ManagedPolicy.fromAwsManagedPolicyName("CloudWatchAgentServerPolicy"),
    iam.ManagedPolicy.fromAwsManagedPolicyName("AWSXrayWriteOnlyAccess")
```
Much easier to add 2 lines like that than exporting the entire policy and putting in a folder.